repose-aggregator/artifacts/valve/src/main/scala/org/openrepose/valve/ReposeJettyServer.scala

```diff
@@ -169,29 +168,11 @@ class ReposeJettyServer(val nodeContext: AbstractApplicationContext,
 
     val contextHandler = new ServletContextHandler()
     contextHandler.setContextPath("/")
-    contextHandler.addServlet(classOf[EmptyServlet], "/*")
-
-    // todo: when switching to the new ReposeFilterChain, replace the previous line with the following line
-    // todo: to wire in the new routing servlet
-    // contextHandler.addServlet(new ServletHolder(appContext.getBean(classOf[ReposeRoutingServlet])), "/*")
-
-    val cll = new ContextLoaderListener(appContext)
-    contextHandler.addEventListener(cll)
+    contextHandler.addServlet(new ServletHolder(appContext.getBean("reposeRoutingServlet").asInstanceOf[HttpServlet]), "/*")
```

Comment:
So initially, we did go with the type. In fact, that is generally my preference.

However, we use the `FakeReposeRoutingServlet` class in testing which is not the same type as the `ReposeRoutingServlet`. So we use the stable identifier "reposeRoutingServlet" rather than the type.

The same applies to the `reposeFilter`.

Of course, we could come up with a different way of testing to enable us to get the bean by type rather than identifier, but this was the quicker solution. Furthermore, this is the way we have been doing things. And of course, if this approach did not work, (nearly) all of our functional tests would fail, so we can be confident that this works even if it may not be the ideal solution. Or rather, this seems like a practical ideal rather than an ideological one.